File: Definitions/Game_Definitions/classes/DoorSystem.lua
---@meta
---@diagnostic disable

---@class DoorSystem : DeviceSystemBase
DoorSystem = {}

---@return DoorSystem
function DoorSystem.new() return end

---@param props table
---@return DoorSystem
function DoorSystem.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function DoorSystem:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function DoorSystem:OnTakeControl(ri) return end

---@return DoorSystemController
function DoorSystem:GetController() return end

---@return DoorSystemControllerPS
function DoorSystem:GetDevicePS() return end

File: DecompiledGameScripts/cyberpunk/network/systems/doorSystem.swift

public class DoorSystem extends DeviceSystemBase {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as DoorSystemController;
    super.OnTakeControl(ri);
  }

  protected const func GetController() -> ref<DoorSystemController> {
    return this.m_controller as DoorSystemController;
  }

  public const func GetDevicePS() -> ref<DoorSystemControllerPS> {
    return this.GetController().GetPS();
  }
}
